bundles/org.eclipse.core.databinding.observable/src/org/eclipse/core/internal/databinding/observable/MapEntryObservableValue.java

```diff
@@ -37,8 +37,8 @@
  */
 public class MapEntryObservableValue<K, V> extends AbstractObservableValue<V> {
 	private IObservableMap<K, V> map;
-	private K key;
-	private Object valueType;
+	private final K key;
+	private final Object valueType;
 
 	private IMapChangeListener<K, V> changeListener = event -> {
 		if (event.diff.getAddedKeys().contains(key)) {
```

Comment:
@jjohnstn the compilation here fails now (see also below warning) with

> The blank final field key may not have been initialized

I think this is because key/value are only initialized in the _constructor_ but already used/referenced in the inner class initialized at _class initialization_ time.... obviously a tricky combination (that would be avoided if we init the `changeListener` in the constructor as well!)

Reply:
Code looks good to me now.